src/lxc/conf.c

```diff
@@ -1587,22 +1577,84 @@ static int lxc_setup_ttydir_console(const struct lxc_rootfs *rootfs,
 	if (ret >= 0)
 		close(ret);
 
-	if (console->master < 0) {
-		INFO("no console");
-		return 0;
-	}
-
-	if (safe_mount(console->name, lxcpath, "none", MS_BIND, 0, rootfs->mount) < 0) {
-		ERROR("failed to mount '%s' on '%s'", console->name, lxcpath);
+	ret = snprintf(path, sizeof(path), "%s/dev/console", rootfs->mount);
+	if (ret < 0 || (size_t)ret >= sizeof(lxcpath))
 		return -1;
+
+	/* When we are asked to setup a console we remove any previous
+	 * /dev/console bind-mounts.
+	 */
+	if (console->path && !strcmp(console->path, "none")) {
```

Comment:
I guess this codepath is something we want to discuss. We can either take any bind-mount for /dev/console absolutely literally and do not move it around in case `lxc.devttydir` is requested or do as I'm doing here, movie it under `lxc.devttydir`.

Reply:
@brauner Hi, i find this PR when i deal with the #2092. 
I do not quit understand why we should do this, if the user specify console->path = none, i think we should do not do anything. The man page of lxc
```
lxc.console.path
Specify a path to a device to which the console will be attached. The keyword 'none' will simply disable the console. Note, when specifying 'none' and creating a device node for the console in the container at /dev/console or bind-mounting the hosts's /dev/console into the container at /dev/console the container will have direct access to the hosts's /dev/console. This is dangerous when the container has write access to the device and should thus be used with caution.
```
I wonder how to  create a device node for the console in the container at /dev/console ? mknod in the container?

Reply:
> @brauner Hi, i find this PR when i deal with the #2092.
I do not quit understand why we should do this, if the user specify console->path = none, i think we should do not do anything. The man page of lxc

Agreed! Please add this to your PR https://github.com/lxc/lxc/pull/2092 if you haven't already. :)

> I wonder how to create a device node for the console in the container at /dev/console ? mknod in the container?

You mean at container runtime or during container setup?
- runtime from inside the container: in order to create a console device node at runtime from inside the container you need to retain `CAP_MKNOD`
- runtime from outside the container: needs `CAP_MKNOD` on the host and involves some tricks with mount namespaces + bind-mounting
- during container setup: needs `CAP_MKNOD` on the host but there's currently no logic to create an actual `/dev/console` device. I'm also not sure that this makes sense or even is secure since `/dev/console` is not namespaced and would thus be just another instance of the hosts `/dev/console` device. That's why we always do bind-mounts of pty devices.